pkgs/applications/misc/obsidian/default.nix

```diff
@@ -14,8 +14,9 @@ let
     desktopName = "Obsidian";
     comment = "Knowledge base";
     icon = "obsidian";
-    exec = "obsidian";
+    exec = "obsidian %u";
     categories = "Office";
+    mimeType = "text/html;x-scheme-handler/obsidian;";
```

Comment:
I believe you only need the `x-scheme-handler/obsidian;` part to handle `obsidian://` scheme.
But not the `text/html;` part.